src/components/pages/Invite/Invite.tsx

```diff
@@ -0,0 +1,158 @@
+import { BackHandler } from "react-native";
+import * as React from "react";
+import { Container } from "../../shared/elements";
+import { InviteCamera } from "./InviteCamera";
+import DropdownAlert from "react-native-dropdownalert";
+import { VideoPreview } from "../Camera/VideoPreview";
+import { MapStateToProps, connect } from "react-redux";
+import {
+  getLoggedInMember,
+  getInviteVideoRef
+} from "../../../store/selectors/authentication";
+import { Member } from "../../../store/reducers/members";
+import { RahaState } from "../../../store";
+import { generateToken } from "../../../helpers/token";
+
+enum InviteStep {
+  CAMERA,
+  VIDEO_PREVIEW,
+  SEND_INVITE
+}
+
+type ReduxStateProps = {
+  loggedInMember?: Member;
+};
+
+type OwnProps = {};
+
+type InviteProps = ReduxStateProps & OwnProps;
+
+type InviteState = {
```

Comment:
Not necessary and potentially overcomplex, but you can potentially make this bullet-proof typewise by defining the states that are allowed explicitly:

```typescript
type BaseInviteState = {
  step: InviteStep
};

// or whatever it should be, when is `verifiedName` not present?
type CameraStepInviteState = BaseInviteState & {
  verifiedName: string
};
type VideoPreviewStepInviteState = CameraStepInviteState & {
 videoUri: string;
}
type CompleteInviteState = VideoPreviewStepInviteState & {
  videoDownloadUrl: string;
}
type InviteState = CameraStepInviteState | VideoPreviewStepInviteState | CompleteInviteState;
```

Reply:
edited it a bit to be even more precise